ratioToSwitchKeyframe: 0.80

transformations:
  - TransformFeatures

readingDataPointsFilters:
  - MaxDistDataPointsFilter:
      maxDist: 7.0
      dim: 2
  - RandomSamplingDataPointsFilter:
      prob: 0.20

keyframeDataPointsFilters:
  - MaxDistDataPointsFilter:
      maxDist: 7.0
      dim: 2
  - SamplingSurfaceNormalDataPointsFilter:
      k: 17

matcher:
  KDTreeMatcher:
    maxDist: 0.1
    epsilon: 0.0

featureOutlierFilters:
  - TrimmedDistOutlierFilter:
      ratio: 0.85

errorMinimizer:
  PointToPlaneErrorMinimizer

transformationCheckers:
  - DifferentialTransformationChecker:
      minDiffRotErr: 0.001
      minDiffTransErr: 0.01
      smoothLength: 4
  - CounterTransformationChecker:
      maxIterationCount: 30
  - BoundTransformationChecker:
      maxRotationNorm: 0.50
      maxTranslationNorm: 0.50

inspector:
  NullInspector

